docker/build/assets.Dockerfile

```diff
@@ -99,7 +99,9 @@ ENV CUDA_QUANTUM_VERSION=$release_version
 
 RUN cd /cuda-quantum && source scripts/configure_build.sh && \
     ## [>CUDAQuantumBuild]
-    CUDAQ_PYTHON_SUPPORT=OFF CUDAQ_WERROR=false \
+    CUDAQ_WERROR=false \
+    CUDAQ_PYTHON_SUPPORT=OFF \
+    CUDAHOSTCXX="$CXX" \
```

Comment:
Doesn't this effectively undo the change in `scripts/build_cudaq.sh`? That is - when building using this Dockerfile, will CUDA be properly detected and supported?

Reply:
The need for this here was the motivation for having originally added that to the build_cudaq.sh script. This variable is indeed needed for the build to work within this environment (otherwise that "check cuda compiler works" checks that cmake runs fail - not sure why the same checks do not run in the dev image...). Conversely, setting it here is fine since we are actually using a supported host compiler (gcc11 rather than gcc12 in the dev image).